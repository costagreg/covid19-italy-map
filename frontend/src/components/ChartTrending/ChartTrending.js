import React, { Component } from 'react'
import { scaleBand, scaleLinear, scale } from 'd3-scale'
import { range, max } from 'd3-array'
import { line, curveCatmullRom } from 'd3-shape'
import { format } from 'date-fns'
import './ChartTrending.scss'

export default class ChartTrending extends Component {
  render() {
    const { xAxis, yAxis, chartWidth, chartHeight, marginBottom } = this.props

    const xAxisScale = scaleLinear()
      .domain([0, xAxis.length - 1])
      .range([0, chartWidth])

    const yAxisScale = scaleLinear()
      .domain([0, max(yAxis)])
      .range([chartHeight - marginBottom, marginBottom])

    const chartLine = line()
      .x((d, i) => xAxisScale(i))
      .y((d) => yAxisScale(d))
      .curve(curveCatmullRom.alpha(0.5))

    const xTicks = xAxisScale.ticks(6).map((d) =>
      xAxis[d] ? (
        <g
          transform={`translate(${xAxisScale(d)},${chartHeight -
            marginBottom})`}
        >
          <text className="chartTrending__xTicksText">
            {format(new Date(+xAxis[d]), 'dd/MM')}
          </text>
          <line
            className="chartTrending__xTicksLine"
            x1="0"
            x1="0"
            y1="0"
            y2="5"
          />
        </g>
      ) : null
    )

    return (
      <div className="chartTrending">
        <svg
          width={chartWidth}
          height={chartHeight}
          viewBox={`0 0 ${chartWidth} ${chartHeight}`}
        >
          <g>
            <line
              className="chartTrending__xAxis"
              x1={0}
              x2={chartWidth}
              y1={chartHeight - marginBottom}
              y2={chartHeight - marginBottom}
            />
          </g>
          <g>
            <path className="chartTrending__line" d={chartLine(yAxis)} />
            {yAxis.map((d, i) => (
              <circle cx={xAxisScale(i)} cy={yAxisScale(d)} r={2} />
            ))}
          </g>
          {xTicks}
        </svg>
      </div>
    )
  }
}

ChartTrending.defaultProps = {
  xAxis: [],
  yAxis: [],
}
