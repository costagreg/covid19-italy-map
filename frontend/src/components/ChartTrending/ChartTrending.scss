.chartTrending{
  &__xAxis{
    stroke:#000;
  }
  &__line{
    stroke: steelblue;
    stroke-width: 2px;
    fill: none;
  }
  &__xTicksText{
    font-size: 12px;
    transform:translate(0,15px)
  }

  &__xTicksLine{
    stroke:#000;
    transform:translate(0,0)
  }

}