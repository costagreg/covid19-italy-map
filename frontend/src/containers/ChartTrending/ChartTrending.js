import React from 'react'
import { Query } from 'react-apollo'
import propTypes from 'prop-types'

import Loader from '../../components/Loader'
import LineChart from '../../components/LineChart'
import { FETCH_LATEST_TREND } from '../../queries'

const ChartTrending = ({ selectedRegion, selectedParam }) => (
  <Query
    query={FETCH_LATEST_TREND}
    variables={{ region: selectedRegion, param: selectedParam }}
  >
    {({ loading, error, data }) => {
      if (loading) return <Loader />
      if (error) return <div>Ops something went wrong</div>

      const { latestTrendParam } = data

      return (
        <div data-testid={`chartTrending-${selectedRegion}`}>
          <LineChart
            xAxis={latestTrendParam.x}
            yAxis={latestTrendParam.y}
            yLabel={selectedParam}
            chartWidth={400}
            chartHeight={200}
            margin={[10, 20, 30, 20]}
          />
        </div>
      )
    }}
  </Query>
)

ChartTrending.propTypes = {
  selectedRegion: propTypes.string.isRequired,
  selectedParam: propTypes.string.isRequired,
}

export default ChartTrending
