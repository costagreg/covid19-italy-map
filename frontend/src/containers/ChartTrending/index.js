export { ChartTrending as default } from './ChartTrending'