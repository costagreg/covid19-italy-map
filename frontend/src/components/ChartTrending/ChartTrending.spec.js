import { render, fireEvent } from '@testing-library/react'
import React from 'react'
import ChartTrending from './ChartTrending'

const initProps = {
}

describe('ChartTrending', () => {
  describe('@render', () => {
    
  })
})
