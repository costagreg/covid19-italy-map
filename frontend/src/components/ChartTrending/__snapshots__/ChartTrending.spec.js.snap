// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartTrending @render renders the ChartTrending without any error 1`] = `
<DocumentFragment>
  <div
    class="chartTrending"
    data-testid="chartTrending"
  >
    <svg
      height="200"
      viewBox="0 0 300 200"
      width="300"
    >
      <g>
        <line
          class="chartTrending__xAxis"
          x1="0"
          x2="300"
          y1="200"
          y2="200"
        />
      </g>
      <g>
        <path
          class="chartTrending__line"
          d="M0,33.33333333333333C0,33.33333333333333,99.77340001969233,2.7151710776023803,150,0C199.7774347633459,-2.690889911713435,300,16.666666666666675,300,16.666666666666675"
        />
        <g
          transform="translate(0,33.33333333333333)"
        >
          <circle
            class="chartTrending__circle"
            r="2"
          />
          <circle
            data-testid="dataPoint"
            fill="transparent"
            r="10"
          />
        </g>
        <g
          transform="translate(150,0)"
        >
          <circle
            class="chartTrending__circle"
            r="2"
          />
          <circle
            data-testid="dataPoint"
            fill="transparent"
            r="10"
          />
        </g>
        <g
          transform="translate(300,16.666666666666675)"
        >
          <circle
            class="chartTrending__circle"
            r="2"
          />
          <circle
            data-testid="dataPoint"
            fill="transparent"
            r="10"
          />
        </g>
      </g>
      <g
        transform="translate(0,200)"
      >
        <text
          class="chartTrending__xTicksText"
        >
          09/04
        </text>
        <line
          class="chartTrending__xTicksLine"
          x1="0"
          y1="0"
          y2="5"
        />
      </g>
      <g
        transform="translate(150,200)"
      >
        <text
          class="chartTrending__xTicksText"
        >
          10/04
        </text>
        <line
          class="chartTrending__xTicksLine"
          x1="0"
          y1="0"
          y2="5"
        />
      </g>
      <g
        transform="translate(300,200)"
      >
        <text
          class="chartTrending__xTicksText"
        >
          11/04
        </text>
        <line
          class="chartTrending__xTicksLine"
          x1="0"
          y1="0"
          y2="5"
        />
      </g>
    </svg>
  </div>
</DocumentFragment>
`;

exports[`ToolTip @render renders the ToolTip without any error 1`] = `
<DocumentFragment>
  <foreignobject
    data-testid="toolTip"
    height="100"
    width="200"
    x="20"
    y="30"
  >
    <div
      class="toolTip"
    >
      <div
        class="toolTip__yValue"
      >
        <b
          class="toolTip__yValueLabel"
        >
          test
        </b>
        :
        <span
          class="toolTip__yValueText"
        >
          60
        </span>
      </div>
      <div
        class="toolTip__xValue"
      >
        12/06/1992
      </div>
    </div>
  </foreignobject>
</DocumentFragment>
`;
